Connect the dots

fureigh opened the pull request:
This PR:
* closes #107,
* closes #110,
* closes #130,
* closes #131,
* closes #133,
* closes #122 and
* closes #123.

(Why so verbose? So GitHub will automatically close each of those issues.)

fureigh committed:
Add TSA confirmation screen template

Still TODO:
* Add buttons
* Consider whether there's a more elegant approach
* Wire it up

...not necessarily in that order.

https://github.com/18F/ffd-info-exchange/issues/110

fureigh committed:
Add passport form content

Confirmation page TK in a separate commit.

https://github.com/18F/ffd-info-exchange/issues/104

fureigh committed:
Add passport confirmation page

Standalone, not yet integrated into the user flow.

https://github.com/18F/ffd-info-exchange/issues/104

fureigh committed:
Add confirmation screen for name change app

https://github.com/18F/ffd-info-exchange/issues/107

fureigh committed:
Make 'bonus services' page a confirmation page

Instead of having it be a step in a multistep form wizard, make it
the confirmation page for the N-400. Next: change its buttons to be
links to the subsequent options.

https://github.com/18F/ffd-info-exchange/issues/122
https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/131

fureigh committed:
Refactoring the way name change works

In service of github.com/18F/ffd-info-exchange/issues/130 etc.

fureigh commented on ffd_info_exchange/uscis/templates/confirmation-name-change.html:
```diff
@@ -0,0 +1,14 @@
+    <p>For status updates on your application, please call the USCIS customer service center at 1-800-666-5713.</p>
+
+    <div class="button_wrapper">
+        <!-- @todo: Get USWDS 'usa-button-small' class to apply. -->
```
@donjo Do you have time to solve this bit?

Context: I changed the markup from a `<button>` to a `<a>` because this page isn't actually a form, but as a result, the `<button class="usa-button-small">` was lost. Applying that class to the `<a>` element doesn't do the trick. I recommend checking the relevant USWDS documentation.

donjo replied:
I'm having some trouble replicating this but I believe you're supposed to use the `usa-button` class too. so `<a class="usa-button usa-button-small">`

fureigh replied:
That did it! Thanks. :)

fureigh committed:
Apply USWDS button styling (name change conf)

h/t @donjo

https://github.com/18F/ffd-info-exchange/issues/130

fureigh committed:
Rework and finish name change flow

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132

Closes https://github.com/18F/ffd-info-exchange/issues/107.

fureigh committed:
Rework passport flow

Form fields are currently ignored.

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132
https://github.com/18F/ffd-info-exchange/issues/104

fureigh committed:
Change get_name_change_form to not accept submissions

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132
https://github.com/18F/ffd-info-exchange/issues/107

fureigh committed:
Rework global entry flow

Form fields are currently ignored.

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132
https://github.com/18F/ffd-info-exchange/issues/110

fureigh committed:
Standardize TSA+Precheck to 'Global Entry'

https://github.com/18F/ffd-info-exchange/issues/110

fureigh committed:
Make passport form fields render

    https://github.com/18F/ffd-info-exchange/issues/130
    https://github.com/18F/ffd-info-exchange/issues/132
    https://github.com/18F/ffd-info-exchange/issues/104

fureigh committed:
Make Global Entry form fields render

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132
https://github.com/18F/ffd-info-exchange/issues/110

fureigh committed:
Make name change form fields render

https://github.com/18F/ffd-info-exchange/issues/130
https://github.com/18F/ffd-info-exchange/issues/132
https://github.com/18F/ffd-info-exchange/issues/107

fureigh commented:
@donjo Ready for review!

donjo reviewed (approved):
seems to be functionally sound. Are we looking to pursue to checkbox style (select which ones you want and receive those optional service forms in sequence) or following the menu of choices style where you pick one and go back and do the others? I'm fine with either and ready to lean towards the one that's easier technically.

If we stick with the way it is now (in this PR) I would probably prioritize some sort of "done" acknowledgement so when users return to that screen they understand what they've already completed. 

We can make further progress on that in a new PR though.

fureigh commented:
@donjo I'm also open to either approach; this is currently the menu of choices style. Great call about prioritizing a "done" acknowledgement. Either direction will require nontrivial changes, so I'll merge this for now, think through which will be simpler to implement, and start a new PR for that work.

fureigh commented:
P.S. Thanks for the review!
